core/arch/x86/decode_table.c

```diff
@@ -1535,13 +1551,18 @@ const instr_info_t * const op_instr[] =
 #define KVd TYPE_K_VEX, OPSZ_4
 #define KVq TYPE_K_VEX, OPSZ_8
 #define KE1b TYPE_K_EVEX, OPSZ_1b
+#define KE2b TYPE_K_EVEX, OPSZ_2b
+#define KE4b TYPE_K_EVEX, OPSZ_4b
 #define KEb TYPE_K_EVEX, OPSZ_1
 #define KEw TYPE_K_EVEX, OPSZ_2
 #define KEd TYPE_K_EVEX, OPSZ_4
 #define KEq TYPE_K_EVEX, OPSZ_8
 #define Ves TYPE_V, OPSZ_16_vex32_evex64
 #define Ved TYPE_V, OPSZ_16_vex32_evex64
 #define Vqq_oq TYPE_V, OPSZ_vex32_evex64
+#define Vdq_e TYPE_V, OPSZ_16_of_vex32_evex64
+#define Vqq_e TYPE_V, OPSZ_32_of_evex64
```

Comment:
Size 'e' means '128, 256, or 512'.  If this is supposed to be `of_64` (see above) this then is hardcoded to 512 so it should 'oq'.  So `Vqq_oq`.  If there is a base then I'm not sure about it b/c sometimes it would be the whole thing and sometimes half?

Reply:
I was expecting a discussion here. Problem is, the letter codes don't differentiate well between "or" and "of". So

Vqq_oq TYPE_V, OPSZ_vex32_evex64 => 32 or 64
Vqq_e TYPE_V, OPSZ_32_of_evex64 => 32 of 64

Sounds like you would name the "of" Vqq_oq. What about the other version then? Yes, it's either 32 "or" 64 (OPSZ_vex32_evex64).

Reply:
Also, we have 'e' for '128, 256, or 512'. We also need something for '256, or 512', e.g. TYPE_V, OPSZ_vex32_evex64. Maybe these letter shortcuts confuse more than they help? It would almost be more clear if we just replaced all letter shorcuts with type, size pairs.

Reply:
The underscore always means "of" so I don't think there's any confusion there.  A single letter is used for an "or", with no underscore.   Vqq_oq does not mean "32 or 64", it means "32 of 64".

If Intel had set the codes for AVX-512 we would just use those: in the past we used their codes and it seemed the right thing to do, to use the "official" codes, and making the tables more compact.  I would say to continue with them

For "256 or 512" -- new code 'f'?

Reply:
Ok^2. There will be the new codes Vdq_f and Vf.

Reply:
The changes are (added to commit message:

#define Vf TYPE_V, OPSZ_vex32_evex64
#define Vdq_f TYPE_V, OPSZ_16_of_32_evex64
#define Vqq_oq TYPE_V, OPSZ_32_of_64
#define Wf TYPE_W, OPSZ_vex32_evex64
#define Hf TYPE_H, OPSZ_vex32_evex64
#define Hdq_f TYPE_H, OPSZ_16_of_32_evex64